moveit_commander/src/moveit_commander/planning_scene_interface.py

```diff
@@ -263,3 +261,47 @@ def get_attached_objects(self, object_ids = []):
             conversions.msg_from_string(msg, ser_aobjs[key])
             aobjs[key] = msg
         return aobjs
+
+    def check_object_does_not_exist(self, object_name, timeout=4):
+        """ Ensure collision object is removed from the planning scene """
+        print("test")
+        return self.wait_for_state_update(object_name, False, False, timeout)
+
+    def check_object_exists_unattached(self, object_name, timeout=4):
+        """ Ensure collision object is added to planning scene """
+        return self.wait_for_state_update(object_name, True, False, timeout)
+
+    def check_object_attached(self, object_name, object_is_known=False, object_is_attached=False, timeout=4):
+        """ Ensure collision object is attached to robot """
+        return self.wait_for_state_update(object_name, False, True, timeout)
+
+    def wait_for_state_update(self, object_name, object_is_known=False, object_is_attached=False, timeout=4):
+        """ Ensuring Collision Updates Are Receieved """
+
+        ## If a node dies before publishing a collision object update message, the message
+        ## could get lost and the object will not appear. To ensure that the updates are
+        ## made, we wait until we see the changes reflected in the
+        ## ``get_attached_objects()`` and ``get_known_object_names()`` lists.
+        ## We call this function after adding, removing, attaching or detaching an object
+        ## in the planning scene. We then wait until the updates have been made or ``timeout``
+        ## seconds have passed
+        start_time = rospy.get_time()
+        current_time = start_time
+        while (current_time - start_time < timeout) and not rospy.is_shutdown():
+          # Test if the object is in attached objects
+          attached_objects = self.get_attached_objects([object_name])
```

Comment:
This calls a ROS service over and over again to check whether a ROS message arrived.
This is horrible design and should not be part of the interface.

Reply:
Do you have a more constructive suggestion? I agree that this is a bad way to do things but I don't see a a better way to ensure that we wait for the planning scene to reflect the desired state

Reply:
Yes, please switch the interfaces in this class to use the `ApplyPlanningScene` service.
This seems the only reasonable option to me.

Reply:
Honestly I agree that using the `ApplyPlanningScene` service would be a cleaner solution here.